### PR TITLE
Add CRUD REST to mod-template-engine

### PR DESCRIPTION
**Pull request by OleksiiKuzminov**
- CRUD REST API for templates was added
- API test was added
- Add description of endpoints to documentation

### CODE REVIEW

File: ramls/examples/templatesCollection.sample [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          "lang": "de",
+          "template": "Hallo ${context.user.name}"
+        }
+      ],
+      "metadata": {
+        "date":"21.11.2010"
+      }
+    }
+  ],
+  "totalRecords": "1"
```
Comment (natali-zaitseva) [inline ramls/examples/templatesCollection.sample]: to make it consistent with templatesCollection.json please change to "totalRecords": 1 where field totalRecords need to have an integer value

File: README.md [hunk 2]
```diff
@@ -12,6 +12,14 @@ text, html, xml, doc, docx etc from the template.
 Generated payload is described by meta information which contain info
```
Comment (natali-zaitseva) [inline README.md]: contains

File: NEWS.md [hunk 3]
```diff
@@ -1,2 +1,16 @@
+## 2018-08-30 v1.0.0-SNAPSHOT
+The module provide storing of templates and access to them through REST API. 
+Templates support internationalization and store a set of templates texts for each language. 
+Also templates contains meta information that describes the type of source document and its scope.
```
Comment (natali-zaitseva) [inline NEWS.md]: please change to 'Also templates contain ...'

File: src/main/java/org/folio/rest/impl/TemplateResourceImpl.java [hunk 4]
```diff
@@ -4,40 +4,263 @@
...
-      if (event.succeeded()) {
-        asyncResultHandler.handle(
-          Future.succeededFuture(
-            Response.ok(event.result().toString()).header(CONTENT_TYPE_HEADER, APPLICATION_JSON).build()
-          ));
+  public void postTemplate(
+    TemplateJson entity,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
```
Comment (natali-zaitseva) [inline src/main/java/org/folio/rest/impl/TemplateResourceImpl.java]: could you please format the code and mark parameters as final

Comment (OleksiiKuzminov) [reply src/main/java/org/folio/rest/impl/TemplateResourceImpl.java]: final is overhead for here. Thanks for code format remark

File: src/test/java/org/folio/rest/impl/RestVerticleTest.java [hunk 5]
```diff
@@ -19,68 +24,167 @@
...
+    .add("html");
+
+  private JsonObject template1 = new JsonObject()
+    .put("lang", "en")
+    .put("template", "Hello ${context.user.name}");
+
+  private JsonObject template2 = new JsonObject()
+    .put("lang", "de")
+    .put("template", "Hallo ${context.user.name}");
+
```
Comment (natali-zaitseva) [inline src/test/java/org/folio/rest/impl/RestVerticleTest.java]: Would it be better to have generalized method like this?
private JsonObject buildLocalizedTemplate(String locale){
JsonObject template = new JsonObject()
    template.put("lang", locale)
    template.put("template", "Hallo ${context.user.name}");
return template;
}

File: src/main/java/org/folio/rest/impl/TemplateResourceImpl.java [hunk 6]
```diff
@@ -4,40 +4,260 @@
...
+      String message = "Template creation internal error";
+      logger.error(message, e);
+      asyncResultHandler.handle(Future.succeededFuture(PostTemplateResponse.withPlainInternalServerError(message)));
+    }
+  }
+
+  @Override
+  public void getTemplate(int length, int start, String sortBy,
+                          String query, Map<String, String> okapiHeaders,
+                          Handler<AsyncResult<Response>> asyncResultHandler,
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/rest/impl/TemplateResourceImpl.java]: Looks like sortBy parameter is never used

File: src/test/java/org/folio/rest/impl/RestVerticleTest.java (hunk 5 above)
Comment (OleksiiKuzminov) [reply src/test/java/org/folio/rest/impl/RestVerticleTest.java]: there are two different templates too. So create a no-reuse method with two params for two objects doesn't looks like needed here

File: NEWS.md [hunk 7]
```diff
@@ -1,2 +1,16 @@
...
+The module provide storing of templates and access to them through REST API. 
+Templates support internationalization and store a set of templates texts for each language. 
+Also templates contain meta information that describes the type of source document and its scope.
+
+CRUD API for templates was added: 
+
+| METHOD                             | DESCRIPTION                                        |
+|------------------------------------|----------------------------------------------------|
+| POST /template                     | Create new template in storage                     |
+| GET /template/id                   | Get template from storage by id                    |
```
Comment (natali-zaitseva) [inline NEWS.md]: I would suggest to change it to 'GET /template/{templateId}' to be more consistent and emphasize that templateId is changeable parameter

**Comment by OleksiiKuzminov** [comment]
@dcrossleyau @kurtnordstrom @jeremythuff Hello all! If there will be no comments, I will be ready to merge this PR

**Comment by OleksiiKuzminov** [comment]
@dcrossleyau @kurtnordstrom @jeremythuff It would be nice if you take a look at the pr, we're going to merge it in 3 days

**Comment by OleksiiKuzminov** [comment]
@kurtnordstrom Thanks for the reviewing! I add changes to the raml file

**Pull request merged**